server/src/internalClusterTest/java/org/opensearch/index/ClusterMergeSchedulerConfigsIT.java

```diff
@@ -0,0 +1,327 @@
+            )
+            .get();
+
+        indexName = "log-myindex-2";
+        createIndex(indexName);
+        ensureYellowAndNoInitializingShards(indexName);
+        ensureGreen(indexName);
+        getIndexResponse = client(clusterManagerName).admin().indices().getIndex(new GetIndexRequest()).get();
+        indicesService = internalCluster().getInstance(IndicesService.class, randomFrom(dataNodes));
+        uuid = getIndexResponse.getSettings().get(indexName).get(IndexMetadata.SETTING_INDEX_UUID);
+        IndexService secondIndexService = indicesService.indexService(new Index(indexName, uuid));
+        assertEquals(40, indexService.getIndexSettings().getMergeSchedulerConfig().getMaxMergeCount());
+        assertEquals(40, secondIndexService.getIndexSettings().getMergeSchedulerConfig().getMaxMergeCount());
+        assertEquals(20, indexService.getIndexSettings().getMergeSchedulerConfig().getMaxThreadCount());
+        assertEquals(20, indexService.getIndexSettings().getMergeSchedulerConfig().getMaxThreadCount());
+        assertEquals(true, indexService.getIndexSettings().getMergeSchedulerConfig().isAutoThrottle());
+        assertEquals(true, secondIndexService.getIndexSettings().getMergeSchedulerConfig().isAutoThrottle());
+
+        int replicas = randomIntBetween(1, Math.max(1, internalCluster().numDataNodes() - 1));
+        // Create index with index level override in settings
+        indexName = "log-myindex-3";
+        createIndex(
+            indexName,
+            Settings.builder()
+                .put(MergeSchedulerConfig.MAX_MERGE_COUNT_SETTING.getKey(), 150)
+                .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, replicas)
+                .build()
+        );
+        ensureYellowAndNoInitializingShards(indexName);
+        ensureGreen(indexName);
+        getIndexResponse = client(clusterManagerName).admin().indices().getIndex(new GetIndexRequest()).get();
+        indicesService = internalCluster().getInstance(IndicesService.class, randomFrom(dataNodes));
+        uuid = getIndexResponse.getSettings().get(indexName).get(IndexMetadata.SETTING_INDEX_UUID);
+        IndexService thirdIndexService = indicesService.indexService(new Index(indexName, uuid));
+
+        assertEquals(150, thirdIndexService.getIndexSettings().getMergeSchedulerConfig().getMaxMergeCount());
+        assertEquals(40, secondIndexService.getIndexSettings().getMergeSchedulerConfig().getMaxMergeCount());
+        assertEquals(40, indexService.getIndexSettings().getMergeSchedulerConfig().getMaxMergeCount());
+        assertEquals(20, indexService.getIndexSettings().getMergeSchedulerConfig().getMaxThreadCount());
+        assertEquals(20, secondIndexService.getIndexSettings().getMergeSchedulerConfig().getMaxThreadCount());
+        assertEquals(3, thirdIndexService.getIndexSettings().getMergeSchedulerConfig().getMaxThreadCount());
```

Comment:
this may not hold true always because this depends on the underlying instance where this test runs. Lets try to get determine the expected value and assert it against the actual value. Apart from this, can we also have a case where 1) we create an index with max thread count set to value "x" and assert that the max merge count is equal to "x+5".